### PR TITLE
Code formatting and pushd compatibility updates

### PR DESCRIPTION
**Pull request by ojarva**
This pull request changes three different things:
- Formatting: added {} for longer if statements, few small spelling fixes and converted all tabs to spaces.
- Renamed x-wns-notificationstatus to x-wns-status. x-wns-notificationstatus header still exists, but it is deprecated.
- Added "shouldDeleteChannel" attribute when appropriate. If error code from notify.microsoft.com is 403, 404 or 410, channel should not be used anymore. This improves API compatibility with for example [MPNS library](https://github.com/jeffwilcox/mpns).

### CODE REVIEW

**Comment by pke** [comment]
good additions. I wonder if they will ever be merged?

**Comment by ojarva** [comment]
I doubt that will ever happen, based on project activity during past few years

**Comment by pke** [comment]
Yeah it seems so. There seem to be other nodejs solutions that handle WNS as well as the Android and Apple use cases anyway. So I guess this project is pretty dead.

**Comment by daohoangson** [comment]
@pke what is the solution that you mentioned?

**Comment by pke** [comment]
This one https://www.npmjs.com/package/cordova-plugin-push-notification or cordova-plugin-push

**Comment by daohoangson** [comment]
@pke from my understanding that package does not implement push mechanism at all. It allows receiving notification on the device only.

**Comment by alex-friedl** [comment]
Hello @ojarva,
I am a new maintainer for this repository and would like to merge your proposed changes. Could you resolve the merge conflicts and answer my one question from the review ? :)

File: test/apis.js [hunk 1]
```diff
@@ -89,13 +89,6 @@ describe('wns.createToast', function () {
 		);
 	});
 
-	it('fails with wrong parameter types', function () {
```
Comment (alex-friedl) [inline test/apis.js]: I know it's been a while, but do you remember why you removed those tests?

Comment (ojarva) [reply test/apis.js]: After three years, I have absolutely no idea.

**Comment by ojarva** [comment]
I can do that next week. However, I'm no longer using this library, so if any major changes in addition to rebase are needed, I recommend someone else adopts this PR.

**Comment by alex-friedl** [comment]
@ojarva how about that rebase? :) no other changes would be required in order for me to merge it.

**Comment by ojarva** [comment]
That rebase would have required quite a bit of work thanks to three year's wait.

**Pull request closed without merging**